lib/rspec/core/bisect/coordinator.rb

```diff
@@ -15,47 +15,40 @@ module Bisect
       #   - Formatters::BisectProgressFormatter: provides progress updates
       #     to the user.
       class Coordinator
-        def self.bisect_with(original_cli_args, configuration, formatter)
-          new(original_cli_args, configuration, formatter).bisect
+        def self.bisect_with(original_cli_args, formatter)
+          new(original_cli_args, formatter).bisect
         end
 
-        def initialize(original_cli_args, configuration, formatter)
+        def initialize(original_cli_args, formatter)
           @shell_command = ShellCommand.new(original_cli_args)
-          @configuration = configuration
-          @formatter     = formatter
+          @notifier      = Bisect::Notifier.new(formatter)
         end
 
         def bisect
-          @configuration.add_formatter @formatter
```

Comment:
so we only used the config to add a formatter. Instead now we just call the formatter directly, right?

So that probably explains why we don't need to register all the events below. This also implies that previously I would have been able to hook in my own formatter to bisect events, where as now I ... can't? Guess it depends where the `formatter` comes from that's passed into the initializer.

All LGTM, I figure you have this in your head, just trying to see if I'm following along :)

Reply:
> so we only used the config to add a formatter. Instead now we just call the formatter directly, right?

We used the config for 2 things:

* To add the formatter
* To access the `reporter`, which we used to publish events to the formatter

Now we just publish events to the formatter directly, without config and the reporter mediating access.

> So that probably explains why we don't need to register all the events below. 

Yep!

> This also implies that previously I would have been able to hook in my own formatter to bisect events, where as now I ... can't?

In theory, yes, you could have, but we did not provide any way for users to specify a custom bisect formatter, so in practice, a user would have had to monkey patch RSpec to make it use their custom formatter for bisect events.  So we aren't breaking any public APIs here.

Reply:
thank you for extra explanation!